oak/common/nonce_generator.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OAK_COMMON_NONCE_GENERATOR_H
+#define OAK_COMMON_NONCE_GENERATOR_H
+
+#include <vector>
+
+#include "asylo/util/statusor.h"
+
+namespace oak {
+
+typedef std::vector<uint8_t> Nonce;
```

Comment:
I did think about that but was not sure whether it is a good idea to use non-type template args. I now did it, but I put the template on the class rather than on the method, as the assumption is that all nonces are equal sized. PTAL.

edit: this actually fails with the following error, do you know why?

```
ERROR: /opt/my-project/examples/hello_world/client/BUILD:21:1: Linking of rule '//examples/hello_world/client:client' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc @bazel-out/k8-fastbuild/bin/examples/hello_world/client/client-2.params

Use --sandbox_debug to see verbose messages from the sandbox
bazel-out/k8-fastbuild/bin/oak/client/_objs/per_call_policy/per_call_policy.pic.o:per_call_policy.cc:function oak::PerCallPolicy::PerCallPolicy(): error: undefined reference to 'oak::NonceGenerator<32ul>::NonceGenerator()'
bazel-out/k8-fastbuild/bin/oak/client/_objs/per_call_policy/per_call_policy.pic.o:per_call_policy.cc:function oak::PerCallPolicy::GetMetadata(grpc::string_ref, grpc::string_ref, grpc::AuthContext const&, std::multimap<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*): error: undefined reference to 'oak::NonceGenerator<32ul>::NextNonce()'
bazel-out/k8-fastbuild/bin/oak/client/_objs/per_call_policy/per_call_policy.pic.o:per_call_policy.cc:function oak::PerCallPolicy::GetMetadata(grpc::string_ref, grpc::string_ref, grpc::AuthContext const&, std::multimap<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*): error: undefined reference to 'std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > oak::NonceToBase64<32ul>(std::array<unsigned char, 32ul> const&)'
collect2: error: ld returned 1 exit status
Target //examples/hello_world/client:client failed to build
```